synthtool/gcp/common.py

```diff
@@ -59,11 +59,37 @@ def _generic_library(self, directory: str, **kwargs) -> Path:
 
         return result
 
-    def py_samples(self, **kwargs) -> Path:
+    def py_samples(self, sample_project_dir: PathOrStr = ".", **kwargs) -> Path:
+        """
+        Determines whether generation is being done in a client library or in a samples
+        folder so it can either generate in the current directory or the client lib's
+        'samples' folder. A custom path for where to generate may also be specified.
+        Renders README.md according to .repo.metadata.json
+
+        Args:
+            sample_project_dir (Union[Path, str]): The directory for samples, where
+            the README is to be generated
+        """
         # kwargs["metadata"] is required to load values from .repo-metadata.json
         if "metadata" not in kwargs:
             kwargs["metadata"] = {}
-        return self._generic_library("python_samples", **kwargs)
+        # load common repo meta information (metadata that's not language specific).
+        if "metadata" in kwargs:
+            self._load_generic_metadata(kwargs["metadata"])
+        in_client_library = Path("samples").exists()
+        if sample_project_dir == ".": #if set to default value
```

Comment:
why do we need this check? Isn't that the default value?

Reply:
I added this for handling the case sample_proj_dir is set by parameter as opposed to through the metadata. I'm open to removing it if that doesn't seem necessary though.

Reply:
Gotcha, makes sense. I think a unit test illustrating the user setting the parameter for the directory would be a good addition. In addition - a test with both the parameter set and repo metadata set would be good (to show which takes precedence).